spec/models/mixins/archived_mixin_spec.rb

```diff
@@ -0,0 +1,83 @@
+RSpec.describe ArchivedMixin do
+  let(:model_name) { :container }
+  let(:klass) { Container }
+  # let(:model) { klass.new }
+  # let(:archived_model) { klass.new() }
+
+  let(:timestamp) { Time.now.utc }
+  let(:active_model) { FactoryBot.create(model_name) }
+  let(:archived_model) { FactoryBot.create(model_name, :deleted_on => timestamp) }
+
+  describe ".archived" do
+    it "fetches archived records" do
+      active_model
+      archived_model
+
+      expect(klass.archived).to eq([archived_model])
+    end
+  end
+
+  describe ".active" do
+    it "fetches active records" do
+      active_model
+      archived_model
+
+      expect(klass.active).to eq([active_model])
+    end
+  end
+
+  describe ".active_for" do
+    it "detects active are valid" do
+      active_model
+      expect(klass.active_for((2.years.ago)..(1.year.ago))).to eq([active_model])
+    end
+
+    it "detects not yet deleted records are valid" do
+      archived_model
+      expect(klass.active_for((2.years.ago)..(1.year.ago))).to eq([archived_model])
+    end
+
+    it "detects already deleted records are not valid" do
```

Comment:
valid/invalid terminology sounds strange here - the records aren't "bad".  I think prefer in range / out of range

```suggestion
    it "detects active records in range" do
      active_model
      expect(klass.active_for((2.years.ago)..(1.year.ago))).to eq([active_model])
    end

    it "detects not yet deleted records are in range" do
      archived_model
      expect(klass.active_for((2.years.ago)..(1.year.ago))).to eq([archived_model])
    end

    it "detects already deleted records are out of range" do
```